### PR TITLE
fix: update deployment run links

### PR DESCRIPTION
**Pull request by nathanyoung**
related: https://github.com/Hyphen/hyphen-app/pull/1335

## Summary

- Route deployment run links through `internal/hyphenApp` instead of formatting the legacy deployment URL inline.
- Prefer project/environment run URLs when deployment metadata is available, falling back to the legacy deployment route when it is not.
- Use the run deployment snapshot when it has route metadata so links survive selected deployment payloads without project/environment details.
- Add unit coverage for alternate IDs, ID fallbacks, legacy fallbacks, and snapshot selection.

## Impact

Screenshot 2026-06-11 at 11 13 51 AM

Deployment commands now return app links that match the project/environment run route when enough metadata is available, while keeping the existing deployment URL fallback for incomplete payloads.

## Validation

- `env HOME=/private/tmp/hx-test-home GOCACHE=/private/tmp/hx-go-cache GOMODCACHE=/Users/nathanyoung/go/pkg/mod go test ./cmd/deploy ./internal/hyphenApp`

Note: `go test ./...` was also attempted. The sandboxed run first hit Go build-cache permissions, then an isolated-home run passed acceptance coverage but failed `internal/oauth` because `TestStartOAuthServer` could not start its local OAuth callback server in this environment.

### CODE REVIEW

File: internal/hyphenApp/hyphenApp.go [hunk 1]
```diff
@@ -26,6 +27,49 @@ func DeploymentLink(organizationId, deploymentId string) string {
...
+	return deployment.Project.ID
+}
+
+func deploymentEnvironmentSegment(deployment models.Deployment) string {
+	if deployment.ProjectEnvironment.AlternateID != "" {
+		return deployment.ProjectEnvironment.AlternateID
+	}
+
+	return deployment.ProjectEnvironment.ID
 }
```
[the remark this answers is not in the document]
Comment (nathanyoung) [reply internal/hyphenApp/hyphenApp.go]: Addressed in `139299a`. I agreed with this feedback because `Deployment` and `DeploymentRun` are non-trivial structs and the helper does not need ownership of copied values. The link helpers now accept pointers, include nil guards, and the unit tests pass pointers plus cover nil inputs.

File: cmd/deploy/deploy.go [hunk 2]
```diff
@@ -342,7 +342,7 @@ func runDeployBody(cmd *cobra.Command, args []string) (map[string]any, error) {
 		return result, fmt.Errorf("failed to create run: %w", err)
 	}
 
-	appUrl := fmt.Sprintf("%s/%s/deploy/%s/runs/%s", apiconf.GetBaseAppUrl(), orgId, selectedDeployment.ID, run.ID)
+	appUrl := hyphenapp.DeploymentRunLinkForRun(orgId, selectedDeployment, *run)
```
[the remark this answers is not in the document]
Comment (nathanyoung) [reply cmd/deploy/deploy.go]: Addressed in `139299a`. I agreed with the nil-risk part of this comment because `CreateRun` returns `*models.DeploymentRun`, and the previous `*run` callsite could panic. The call now passes `&selectedDeployment` and `run`; I also added an explicit nil-run error in `deploy` because later code still needs `run.ID` and the run stream.
